### DISCUSSION TITLE
Fuji PFX 4 PID Setting

### DISCUSSION THREAD

**Discussion by Dgocof** [artisan-roaster-scope/artisan#1275] [General]
I have a question about where Artisan is getting my PID settings from. I would like Artisan to get them from the Fuji.
Something doesn't seem correct with my setup. In device assignment should the PID be tagged instead of Meter. I change the numbers in the Control window PID and press set. The new numbers are written to the Fuji but don't seem to change the trajectory of the profile  much in the beginning and then throughout the profile. I've used P numbers from .005 to 200 on different roasts. I'm trying to fine tune my PID settings to follow the back round profile a little closer. It can run over and under 2-5 degrees after the profile catches up around 260 degrees. Not sure if I am on the right track.

![Device Assignment](https://github.com/artisan-roaster-scope/artisan/assets/117209911/07bf7315-b050-49af-a94f-5d09bf9e8555)
![Port Config](https://github.com/artisan-roaster-scope/artisan/assets/117209911/b9172e66-4beb-416b-9073-50f7dfdce544)
![Device Assignment2](https://github.com/artisan-roaster-scope/artisan/assets/117209911/9851f197-cbc0-46b9-8f14-e830e45cd193)

**Comment by bigl00z3** [comment]
Dgocof,

Looking at your screenshots I don't see anything that looks wrong to me (maybe the artisan-settings files could help but I think you have documented everything required).

If you say that no mater what you yet in Artisan as P, I or D values and the curve looks the same I wonder if you are not editing in one of the pallete and using an other one.
You can check the PXF4 user manual and Modbus communication manual to find in which pallet you are working.
The Fuji PXF 4 [user manuals are available here](https://www.fujielectric.com/products/instruments/products/controller/PXF.html)

<img width="595" alt="image" src="https://github.com/artisan-roaster-scope/artisan/assets/8886922/9c002951-5468-438b-a611-1251392ff605">

On my PXF4, I have to use Pallet n° 2 if I want to control remotely (as i is "LoCl" / local):

<img width="837" alt="image" src="https://github.com/artisan-roaster-scope/artisan/assets/8886922/dd4a9cfd-3a06-474b-9f0a-dd5c8c45f893">

To change palette you have to use the below menu:

<img width="769" alt="image" src="https://github.com/artisan-roaster-scope/artisan/assets/8886922/0d98b10a-00f7-4a7e-a406-3b69803c4f00">

And when you edit, let's say "P", go in your PID menu to double check that the value has really changed in the Palette you have been planning to use.

I hope this helps

Nico
